main.c

```diff
@@ -216,16 +216,19 @@ static int recvfd(int sock)
 	return fd;
 }
 
-static int parent(int sock, int exit_fd, unsigned int mtu, bool enable_ipv6)
+static int parent(int sock, int exit_fd, struct slirp_config *cfg)
 {
 	int rc, tapfd;
 	if ((tapfd = recvfd(sock)) < 0) {
 		return tapfd;
 	}
 	fprintf(stderr, "received tapfd=%d\n", tapfd);
 	close(sock);
-	printf("starting slirp, MTU=%d\n", mtu);
-	if ((rc = do_slirp(tapfd, exit_fd, mtu, enable_ipv6)) < 0) {
+	printf("starting slirp, MTU=%d\n", cfg->mtu);
+	if (!cfg->no_host_loopback) {
+		printf("WARNING: 127.0.0.1:* on the host is accessible as 10.0.2.2 (set --no-host-loopback to prohibit connecting to 127.0.0.1:*)\n");
+	}
```

Comment:
If we're going to do it this way, I'd suggest that security should be **opt-out** (in other words, we should make it `--host-loopback` and by default disable it). I doubt anyone is depending on our behaviour.

Reply:
The benchmark suite of slirp4netns itself and rootlesskit depends on the 10.0.2.2.
So the default value won't change on slirp4netns side.

However, downstream projects such as podman/docker/usernetes may launch slirp4netns with `--no-host-loopback` as the default.

Reply:
Why don't we just make our own test suite and `rootlesskit` pass it? If this is a security fix, I don't see why we should default to being insecure -- not to mention that downstreams will need to check whether their `slirp4netns` version supports the flag (on the other hand we'd only need to do this for `rootlesskit`).